src/LibraryManager.Vsix/Json/Completion/VersionCompletionEntry.cs

```diff
@@ -0,0 +1,44 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using Microsoft.VisualStudio.Imaging.Interop;
+using Microsoft.VisualStudio.Language.Intellisense;
+using Microsoft.Web.Editor.Completion;
+using Microsoft.VisualStudio.Text;
+using System.Diagnostics;
+using Microsoft.Web.LibraryManager.Providers.Unpkg;
+
+namespace Microsoft.Web.LibraryManager.Vsix
+{
+    internal class VersionCompletionEntry : SimpleCompletionEntry
+    {
+        internal string VersionText { get; private set; }
+
+        public VersionCompletionEntry(string displayText, string insertionText, string description, ImageMoniker moniker, ITrackingSpan span, IIntellisenseSession session, int specificVersion)
+            : base(displayText, insertionText, description, moniker, span, session, specificVersion)
+        {
+            VersionText = displayText;
+        }
+
+        protected override int InternalCompareTo(CompletionEntry other)
+        {
+            VersionCompletionEntry otherEntry = other as VersionCompletionEntry;
+
+            if (otherEntry == null)
+            {
+                return 1;
+            }
+
+            if (!string.IsNullOrEmpty(VersionText))
+            {
+                string otherVersionText = otherEntry.VersionText;
+                SemanticVersion selfSemanticVersion = SemanticVersion.Parse(VersionText);
```

Comment:
>SemanticVersion.Parse(VersionText) [](start = 54, length = 34)

Is InternalCompareTo called often? Is SemanticVersion.Parse at all expensive? If the answer to either of these is "yes", then consider moving this to a member variable.

Reply:
I think it's called often. I would obtain this value in the construct function.

---
In reply to: [198905536](https://github.com/aspnet/LibraryManager/pull/258#discussion_r198905536) [](ancestors = 198905536)